products/memgraph/upcoming/reference_guide/replication.md

```diff
@@ -0,0 +1,116 @@
+### How to Setup a Memgraph Cluster with Replication?
+
+Replication configuration is done primarily through openCypher commands. This
+allows the cluster to be dynamically rearranged (new leader election, addition
+of a new replica, etc.).
+
+Each Memgraph instance when first started will be a main. You have to change
+the role of all replica nodes using the following openCypher query before you
+can enable replication on the main:
+
+```plaintext
+SET REPLICATION ROLE TO (MAIN|REPLICA) WITH PORT <port_number>;
+```
+
+Note that the "WITH PORT <port_number>" part of the query sets the replication port,
+but it applies only to the replica. In other words, if you try to set the
+replication port as the main, a semantic exception will be thrown.
```

Comment:
Semantic exception doesn't mean much to the user I think? Maybe just note that it's not valid and the query will fail.